decorators_library/decorators.py

```diff
@@ -1 +1,132 @@
-# implement your decorators here.
+import time
+import inspect
+from .exceptions import TimeoutError
+from logging import getLogger
+
+class Timeout(object):
+    def __init__(self, func, max_time):
+        self.func = func
+        self.max_time = max_time
+    
+    def __call__(self, *args, **kwargs):
+        start = time.time()
+        return_val = self.func(*args, **kwargs)
+        end = time.time()
+        
```

Comment:
From what I can tell, this doesn't work.

What this does:
t1 = time now.
function happens
t2 = time now

if t2  - t1 > max time allowed, then you return an error

what we're trying to do is 
func happens - while func is happening, if max time is reached, we stop running func

it works in our test cases because it's like 2 seconds long max, but what if the func was 2 hours? it wouldn't help if the 2 hour long func called and then you said "oh, sorry, needed to time you out 1 hour and 55 minutes ago."